splink/comparison_level_library.py

```diff
@@ -68,3 +70,58 @@ def create_sql(self, sql_dialect: SplinkDialect) -> str:
 
     def create_label_for_charts(self) -> str:
         return f"Levenshtein distance of {self.col_name} <= {self.distance_threshold}"
+
+
+class DatediffLevel(ComparisonLevelCreator):
+    def __init__(
+        self,
+        col_name: str,
+        date_threshold: int,
+        date_metric: str = "day",  ##TODO: Lock down to sqlglot supported values
+        cast_strings_to_date: bool = False,
+        date_format: str = "%Y-%m-%d",
+    ):
+        """A comparison level using a date difference function
+
+        e.g. abs(date_diff('day', "mydate_l", "mydate_r")) <= 2  (duckdb dialect)
+
+        Args:
+            col_name (str): Input column name
+            date_threshold (int): The threshold for the date difference
+            date_metric (str): The unit of time ('day', 'month', 'year')
+                for the threshold
+            cast_strings_to_date (bool): Whether to cast string columns to date format
+            date_format (str): The format of the date string
+        """
+        super().__init__(col_name)
```

Comment:
I know the code is largely provisional, but if we end up keeping `ComparisonLevelCreator`'s constructor in its current form, I'd argue we might as well just write:
`self.col_name = col_name` here as it's clearer what's happening.

Reply:
I think Andy's probably right on this one, and it gives us more flexibility to call super. For example, we'll soon need to figure out what to do with levels with multiple input columns (e.g. columns reversed)